src/ILCompiler.Compiler/src/Compiler/MethodExtensions.cs

```diff
@@ -77,13 +77,11 @@ public static SpecialMethodKind DetectSpecialMethodKind(this MethodDesc method)
         {
             if (method.IsPInvoke)
             {
-                // Marshalling is never required for pregenerated interop code
-                if (Internal.IL.McgInteropSupport.IsPregeneratedInterop(method))
-                {
-                    return SpecialMethodKind.PInvoke;
-                }
-
-                if (!Internal.IL.Stubs.PInvokeMarshallingILEmitter.RequiresMarshalling(method))
+                // Only treat the synthetic target methods as real PInvoke.
```

Comment:
(I will have several uses for this helper in the GC transitions change that I am working on.)